### ISSUE TITLE
Crash when /tmp is full

### ISSUE THREAD

**Issue by piegamesde** [duplicati/duplicati#3281]
- [X] I have searched open and closed issues for duplicates.

([Related](https://forum.duplicati.com/t/backblaze-b2-local-tmp-files-fills-up-entire-disk-ramdrive-with-2-0-3-6/3477/9))
----------------------------------------

## Environment info

- **Duplicati version**: 2.0.3.7_canary_2018-06-17
- **Operating system**: Arch Linux
- **Backend**: SFTP

## Description

I have mounted my `/tmp` folder into RAM for performance reasons. During a backup, Duplicati managed to fill it up entirely and then crash (see the bug report linked above).

## Steps to reproduce

1. Have little, limited space on `/tmp`
2. Reproduce the temp-filling bug (probably using large backup files)
3. Wait until it crashes

- **Actual result**:
Duplicati crashes, stopping the server and leaving a lot of waste in `/tmp`

- **Expected result**:
Duplicati stops the backup gracefully with an exception, deletes all of its temp junk and goes on.

## Debug log

```
Jun 19 01:36:16 mono[910]: [ERROR] FATAL UNHANDLED EXCEPTION: System.IO.IOException: Disk full. Path /tmp/dup-53181473-7f91-4357-9801-84cca799d004
Jun 19 01:36:16 mono[910]:   at System.IO.FileStream.WriteInternal (System.Byte[] src, System.Int32 offset, System.Int32 count) [0x00077] in <1050938ab6ed42698bd>
Jun 19 01:36:16 mono[910]:   at System.IO.FileStream.Write (System.Byte[] array, System.Int32 offset, System.Int32 count) [0x00090] in <1050938ab6ed42698bde16066>
Jun 19 01:36:16 mono[910]:   at Duplicati.Library.Utility.Utility.CopyStream (System.IO.Stream source, System.IO.Stream target, System.Boolean tryRewindSource, S>
Jun 19 01:36:16 mono[910]:   at Duplicati.Library.Utility.Utility.CopyStream (System.IO.Stream source, System.IO.Stream target) [0x00000] in <5c22135b7083459abdb>
Jun 19 01:36:16 mono[910]:   at Duplicati.Library.Encryption.GPGEncryption.Runner (System.Object x) [0x00007] in <e9a920c08a254e69b3e84390561f9dce>:0
Jun 19 01:36:16 mono[910]:   at System.Threading.ThreadHelper.ThreadStart_Context (System.Object state) [0x00025] in <1050938ab6ed42698bde160668e4367c>:0
Jun 19 01:36:16 mono[910]:   at System.Threading.ExecutionContext.RunInternal (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallba>
Jun 19 01:36:16 mono[910]:   at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callb>
Jun 19 01:36:16 mono[910]:   at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callb>
Jun 19 01:36:16 mono[910]:   at System.Threading.ThreadHelper.ThreadStart (System.Object obj) [0x0000f] in <1050938ab6ed42698bde160668e4367c>:0
```

**Comment by JonMikelV** [comment]
I believe this is also being discussed on the forum:
https://forum.duplicati.com/t/duplicati-temp-dup-xxxx-files-not-being-deleted/3614

**Comment by gdlmx** [comment]
The same problem occurs on Windows 10 with version 2.0.3.3. I had a very bad experience about this: After crashing in the middle of a backup due to the above error, I used the `--tempdir` option to point to somewhere with larger space and  continued the backup. Finally I ended up with a corrupted `dlist.zip.aes` file!! However there is no warning about it upon following operations. Therefore I didn't notice any corruption at that time, very likely because all the GUI and CLI commands relied on the local sqlite database cache which contained the correct file list. Later I lost the old system together with that sqlite cache, and then I finally realized all my files disappear from the restore list. But it's already too late...
